### PR TITLE
Add declarations interface to SymbolManager

### PR DESCRIPTION
**Pull request by ajreynol**
Partially addresses https://github.com/cvc5/cvc5/discussions/10454.

### CODE REVIEW

File: include/cvc5/cvc5_parser.h [hunk 1]
```diff
@@ -77,6 +77,24 @@ class CVC5_EXPORT SymbolManager
...
   const std::string& getLogic() const;
 
+  /**
+   * Get the list of sorts that have been declared via `declare-sort` commands.
+   * These are the sorts that are printed as part of a response to a
+   * `get-model` command.
+   *
+   * @return The declared sorts.
+   */
+  std::vector<Sort> getModelDeclaredSorts() const;
```
Comment (aniemetz) [inline include/cvc5/cvc5_parser.h:87]: Would it make sense to use `getDeclaredSorts()` and `getDeclaredTerms()` here? Users can query this independently from a model or the status of the input.

Comment (ajreynol) [reply include/cvc5/cvc5_parser.h:87]: Sure, this is probably a better name.

File: src/api/cpp/cvc5_parser.cpp [hunk 2]
```diff
@@ -86,6 +86,24 @@ const std::string& SymbolManager::getLogic() const
   CVC5_API_TRY_CATCH_END;
 }
 
+std::vector<Sort> SymbolManager::getDeclaredSorts() const
+{
+  CVC5_API_TRY_CATCH_BEGIN;
+  //////// all checks before this line
+  return d_sm->getModelDeclareSorts();
```
Comment (aniemetz) [inline src/api/cpp/cvc5_parser.cpp:93]: Do we want to rename this internally, too?

Comment (ajreynol) [reply src/api/cpp/cvc5_parser.cpp:93]: Sure.

**Review by aniemetz — APPROVED** [review]

File: test/unit/api/python/test_symbol_manager.py [hunk 3]
```diff
@@ -43,3 +43,8 @@ def test_get_logic(solver):
         sm.getLogic()
     parse_and_set_logic(solver, sm, "QF_LIA")
     assert sm.getLogic() == "QF_LIA"
+
+def test_get_model_declared_terms_and_sorts(solver):
```
Comment (aniemetz) [inline test/unit/api/python/test_symbol_manager.py:47]:
```suggestion
def test_get_declared_terms_and_sorts(solver):
```

File: test/unit/api/cpp/input_parser_black.cpp [hunk 4]
```diff
@@ -303,5 +303,28 @@ TEST_F(TestInputParserBlack, incrementalSetString)
   ASSERT_EQ(out.str().empty(), true);
 }
 
+TEST_F(TestInputParserBlack, modelDeclares)
```
Comment (aniemetz) [inline test/unit/api/cpp/input_parser_black.cpp:306]:
```suggestion
TEST_F(TestInputParserBlack, getDeclaredSortsAndTerms)
```

File: test/unit/api/java/InputParserTest.java [hunk 5]
```diff
@@ -224,4 +224,24 @@ void multipleParsers() throws CVC5ApiException
     assertThrows(CVC5ApiException.class,
         () -> p5.setIncrementalStringInput(InputLanguage.SMT_LIB_2_6, "input_parser_black"));
   }
+  @Test
+  void modelDeclares()
```
Comment (aniemetz) [inline test/unit/api/java/InputParserTest.java:228]:
```suggestion
  void getDeclaredTermsAndSorts()
```

File: test/unit/api/python/test_input_parser.py [hunk 6]
```diff
@@ -191,3 +191,22 @@ def test_multiple_parsers(solver):
     p5 = InputParser(s4, sm)
     with pytest.raises(RuntimeError):
         p5.setIncrementalStringInput(cvc5.InputLanguage.SMT_LIB_2_6, "test_input_parser")
+
+
+def test_model_declares(solver):
```
Comment (aniemetz) [inline test/unit/api/python/test_input_parser.py:196]:
```suggestion
def test_get_declared_terms_and_sorts(solver):
```

**Pull request merged**